book/src/integrations/oauth2.md

```diff
@@ -458,58 +458,78 @@ Each client has unique signing keys and access secrets, so this is limited to ea
 
 ## WebFinger
 
+> [!NOTE]
+>
+> WebFinger support requires Kanidm v1.5.1 or later.
+
 [WebFinger](https://datatracker.ietf.org/doc/html/rfc7033) provides a mechanism
-for discovering information about people or other entities. It can be used by an
-identity provider to supply OpenID Connect discovery information.
+for discovering information about entities at a well-known URL
+(`http://example.com/.well-known/webfinger`).
+
+It can be used by a WebFinger client to
+[discover the OIDC issuer URL](https://datatracker.ietf.org/doc/html/rfc7033#section-3.1)
+of an identity provider from the hostname alone, and seems to be intended to
+support dynamic client registration flows for large public identity providers.
 
-Kanidm provides
-[an Identity Provider Discovery for OIDC URL](https://datatracker.ietf.org/doc/html/rfc7033#section-3.1)
-response to all incoming WebFinger requests, using a user's SPN as their account
-ID. This does not match on email addresses as they are not guaranteed to be
-unique.
+Kanidm v1.5.1 and later can respond to WebFinger requests, using a user's SPN as
+the account (eg: `user@idm.example.com`). This *does not* match on email
+addresses, because they are not required by Kanidm nor guaranteed to be unique.
```

Comment:
```suggestion
the account (eg: `user@idm.example.com`). This may not match their email
address possibly leading to problems in the client. Because the email attribute is not required by Kanidm nor guaranteed to be unique, it is up to the administrator to ensure that it matches the spn if the service requires that.
```

Reply:
I did some more reading, I was incorrect to suggest that's because of how Kanidm works with email addresses. [`acct` is explicitly "maybe not email address"](https://datatracker.ietf.org/doc/html/rfc7565):

> It is not assumed that an entity will necessarily be able to interact with a user's account using any particular application protocol, such as email

However, I strongly suspect that some clients will assume a vaguely email address shaped account ID *is* a valid email address, so I'll update this.

Reply:
Now reworded in light of the above.